src/assets/js/admin/promotions.js

```diff
@@ -0,0 +1,155 @@
+jQuery( document ).ready( function( $ ) {
+
+
+	$.JiltPromotions = {}
+
+
+	/**
+	 * Jilt Plugin Installation modal handler.
+	 *
+	 * @since 1.1.0-dev.1
+	 */
+	$.JiltPromotions.InstallPluginModal = class InstallPluginModal {
+
+
+		/**
+		 * Constructor.
+		 *
+		 * @since 1.1.0-dev.1
+		 *
+		 * @param {object} options
+		 */
+		constructor( options ) {
+
+			this.options = {
+				messageID : options.messageID || '',
+				target    : options.target    || '',
+				onClose   : options.onClose   || '',
+			};
+
+			if ( '' === this.options.messageID ) {
+				console.log( 'InstallPluginModal: missing messageID' )
+				return;
+			}
+
+			if ( '' === this.options.target ) {
+				this.options.target = 'sv-wc-jilt-promotions-' + this.options.messageID + '-modal';
+			}
+
+			this.initialize();
+			this.open();
+		}
+
+
+		/**
+		 * Initializes the modal.
+		 *
+		 * @since 1.1.0-dev.1
+		 */
+		initialize() {
+
+			// when the install button is clicked
+			$( '#sv-wc-jilt-install-button-install' ).on( 'click', function( event ) {
+				InstallPluginModal.onInstall( event )
+			} );
+
+			// when the newly opened modal is closed
+			$( '#sv-wc-jilt-install-modal .modal-close' ).on( 'click', function( event ) {
+				InstallPluginModal.onClose( event );
+			} );
+		}
+
+
+		/**
+		 * Opens the modal.
+		 *
+		 * @since 1.1.0-dev.1
+		 */
+		open() {
+
+			// ensures there are no other modals opened
+			$( '#wc-backbone-modal-dialog .modal-close' ).trigger( 'click' );
+
+			new $.WCBackboneModal.View( {
+				target: this.options.target,
+			} );
+		}
+
+
+		/**
+		 * Fires when the user clicks on the install button from the modal prompt.
+		 *
+		 * @since 1.1.0-dev.1
+		 *
+		 * @param {_Event} event install click event
+		 */
+		static onInstall( event ) {
```

Comment:
We need this to be instance methods to be able to access the objects options:

```suggestion
		onInstall( event ) {
```